tests/api/test_api.py

```diff
@@ -113,18 +113,28 @@ def test_events(self, app):
         times.
         """
 
+        def get_next_event(stream):
+            data = b""
+            while True:
+                line = stream.raw.readline()
+                if line == b"\n":
+                    break
+                assert line.startswith(b"data: ")
+                data += line[6:-1]
```

Comment:
This `while` loop seems strange, as it merges several lines starting with `data: ` until an empty line is read, and then callers call `json.loads(data)` on the result. If this is the expected behavior, this should be documented as a comment.

Moreover if all callers are expected to call `json.loads`, it would make sense to move such calls inside `get_next_event`.